src/app/pages/submission-pages/overview-page/overview-page.component.ts

```diff
@@ -292,78 +268,69 @@ export class OverviewPageComponent implements OnInit {
     );
   }
 
-  isDataSubTypeSelected(dataSubTypeKey) {
-    if(this.selectedDataSubType.indexOf(dataSubTypeKey) > -1) {
-      return true;
-    }
-
-    return false;
-  }
-
-  onSelectDataType(event, dataType) {
-    this.selectedDataSubType = [];
-    this.selectedDataType = dataType;
+  onSelectSubmissionPlan(event, submissionPlan) {
+    this.selectedSubmissionPlan = submissionPlan;
 
     this.overviewForm.patchValue({
-      dataType: this.selectedDataType,
-      dataSubType: this.selectedDataSubType,
-    });
-  }
-
-  onSelectSubDataType(event, dataSubType) {
-    let itemIndex = this.selectedDataSubType.indexOf(dataSubType);
-
-    if(itemIndex > -1) {
-      this.selectedDataSubType.splice(itemIndex, 1);
-    }
-    else {
-      this.selectedDataSubType.push(dataSubType);
-    }
-
-    this.overviewForm.patchValue({
-      dataType: this.selectedDataType,
-      dataSubType: this.selectedDataSubType,
+      submissionPlan: this.selectedSubmissionPlan
     });
   }
 
   onChangeField(fieldName: string) {
-    let message = "You might loss uploaded data and samples if you have changed this field value. Are you sure?";
+    const message = 'You might loss uploaded data and samples if you have changed this field value. Are you sure?';
 
-    if(this.activeSubmission && !confirm(message)) {
+    if (this.activeSubmission && !confirm(message)) {
       return;
     }
 
-    if(fieldName == 'human') {
+    if (fieldName === 'human') {
       delete this.savedHuman;
     }
 
-    if(fieldName == 'controlled') {
+    if (fieldName === 'controlled') {
       delete this.savedControlled;
     }
 
-    if(fieldName == 'dataType') {
-      this.selectedDataType = this.savedDataType;
-      this.savedDataType = "";
-      this.selectedDataSubType = this.savedDataSubType;
-      this.savedDataSubType = [];
+    if (fieldName === 'submissionPlan') {
+      this.selectedSubmissionPlan = this.savedSubmissionPlan;
+      this.savedSubmissionPlan = '';
     }
   }
 
   onUpdateField(fieldName) {
-    if(fieldName == "human") {
+    if (fieldName === 'human') {
       this.savedHuman = this.overviewForm.value[fieldName];
     }
 
-    if(fieldName == "controlled") {
+    if (fieldName === 'controlled') {
       this.savedControlled = this.overviewForm.value[fieldName];
     }
   }
 
   isTabsDisabled() {
-    if(!this.activeSubmission) {
+    if (!this.activeSubmission) {
       return true;
     }
 
     return false;
   }
+
+  private createRequestBodyAndParams() {
+    // tslint:disable-next-line:prefer-const
```

Comment:
Sorry do we need this comment? Also it would be great if you add a condition or validation in case of this.overviewForm is not exist to avoid any run time errors.

Reply:
I would remove that `lint` comment and change `let overviewData` to `const overviewData`. I think we should do more linting of our code. It really helps to improve the quality!

Reply:
```

Reply:
It is fine with me, I will rewrite to your one. I like it.